app/atomic_svc.py

```diff
@@ -0,0 +1,240 @@
+
+        if not path_yaml:
+            path_yaml = os.path.join(self.repo_dir, 'atomics', '**', 'T*.yaml')
+
+        for filename in glob.iglob(path_yaml):
+            for entries in BaseWorld.strip_yml(filename):
+                for test in entries['atomic_tests']:
+                    try:
+                        incr_ingested, incr_total = await self._save_ability(entries, test)
+                        self.at_ingested += incr_ingested
+                        self.at_total += incr_total
+                    except Exception as e:
+                        self.log.debug('ERROR:', filename, e)
+                        self.errors += 1
+
+        errors_output = f' and ran into {self.errors} errors' if self.errors else ''
+        self.log.debug(f'Ingested {self.at_ingested} abilities (out of {self.at_total}) from Atomic plugin{errors_output}')
+
+    """ PRIVATE """
+
+    def _gen_single_match_tactic_technique(self, mitre_json):
+        '''
+        Generator parsing the json from 'enterprise-attack.json',
+        and returning couples (phase_name, external_id)
+        '''
+        for obj in mitre_json.get('objects'):
+            if not obj.get('type') == 'attack-pattern':
+                continue
+            for e in obj.get('external_references'):
+                if not e.get('source_name') == 'mitre-attack':
+                    continue
+                external_id = e.get('external_id')
+                for kc in obj.get('kill_chain_phases'):
+                    if not kc.get('kill_chain_name') == 'mitre-attack':
+                        continue
+                    phase_name = kc.get('phase_name')
+                    yield (phase_name, external_id)
+
+    async def _populate_dict_techniques_tactics(self, entreprise_attack_path=None):
+        '''
+        Populate internal dictionary used to match techniques to corresponding tactics.
+        By default, use the file 'enterprise-attack.json' located in the Atomic Red Team repository.
+        You can use a specific version of this file by passing its path with the `entreprise_attack_path`
+        parameter.
+        '''
+        self.technique_to_tactics = defaultdict(list)
+        if not entreprise_attack_path:
+            entreprise_attack_path = os.path.join(self.repo_dir, 'atomic_red_team', 'enterprise-attack.json')
+
+        with open(entreprise_attack_path, 'r') as f:
+            mitre_json = json.load(f)
+
+        for phase_name, external_id in self._gen_single_match_tactic_technique(mitre_json):
+            self.technique_to_tactics[external_id].append(phase_name)
+
+    def _handle_attachment(self, attachment_path):
+        # attachment_path must be a POSIX path
+        payload_name = os.path.basename(attachment_path)
+        # to avoid collisions between payloads with the same name
+        with open(attachment_path, 'rb') as f:
+            h = hashlib.md5(f.read()).hexdigest()
+        payload_name = h[:6] + '_' + payload_name
```

Comment:
6 is a magic number. let's use a variable instead

Reply:
fixed in 5b66cb5